tests/ui/closures/2229_closure_analysis/unresolvable-upvar-issue-87987.rs

```diff
@@ -0,0 +1,42 @@
+//! When a closure syntactically captures a place, but doesn't actually capture
+//! it, make sure MIR building doesn't ICE when handling that place.
```

Comment:
pre-existing but if it's really about ICEs then this most likely doesn't need to be a run-pass test.

Reply:
You might be right, but one thing that makes me hesitant is that non-capturing array patterns have really bad test coverage in the test suite overall, in terms of invoking the later parts of the compiler.

So having this test run and not crash feels like it does provide some tiny bit of value? Not sure.

Reply:
We're not checking the output so we'd only catch some catastrophic runtime failure? That is, this could be a build-pass and still add coverage to the patterns' codegen.